File: examples/python/_7_Demo/subscriber.py
from SharedPubSub import *
import os
from time import sleep
## Watchout :
# If readWait() without timeout is in the main thread,
# Use ctrl+\ instead of ctrl+c to exit.

subscriber = Subscriber_int("Demo","DemoSubscriberPy",True)

while(True):
    os.system('clear')
    for i in range(0,8):
        value = subscriber.readWait()
        print("SUBSCRIBER PY:",(f"{value} readWait()" ) if value!=None else "readWait() No value in queue")
    sleep(3)
    for i in range(0,3):
        value = subscriber.readValue()
        print("SUBSCRIBER PY:",f"{value} readValue()" )
        sleep(1)
    
    print("SUBSCRIBER PY: waitForNotify()")
    subscriber.waitForNotify()
    
    value = subscriber.readValue()
    print("SUBSCRIBER PY:",f"{value} readValue()" )
    sleep(5)

File: examples/nodejs/_7_Demo/subscriber.js
const SharedPubSub = require('SharedPubSub');
const sleep = (ms) => new Promise(resolve => setTimeout(resolve, ms));

const subscriber = new SharedPubSub.Subscriber_int("Demo", "DemoSubscriberJS", true);
var value;

async function loop() {
    while(1){
        console.clear();

        for(let i=0;i<8;i++){
            value = await subscriber.readWaitAsync();
            if(value !== null && value !== undefined){
                console.log(`SUBSCRIBER JS : ${value} readWaitAsync()`);
            }
            else{
                console.log("SUBSCRIBER JS : No value in queue");
            }
        }
        await sleep(3000);
        for(let i=0;i<3;i++){
            value = subscriber.readValue();
            console.log(`SUBSCRIBER JS : ${value} readValue()`);
            await sleep(1000);
        }

        console.log(`SUBSCRIBER JS : waitForNotifyAsync()`);
        await subscriber.waitForNotifyAsync()
        value = subscriber.readValue();
        console.log(`SUBSCRIBER JS : ${value} readValue()`);
        await sleep(5000);
    }
}

Promise.all([
    loop().catch(console.error),
])